src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -924,9 +924,57 @@ private static Symbol GetNearestOtherSymbol(ConsList<TypeSymbol> list, TypeSymbo
             Debug.Assert(type.IsInterface);
 
             LookupMembersWithoutInheritance(current, type, name, arity, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics, basesBeingResolved);
-            if ((options & (LookupOptions.NamespaceAliasesOnly | LookupOptions.NamespacesOrTypesOnly)) == 0 && !originalBinder.InCrefButNotParameterOrReturnType)
+            if ((options & LookupOptions.NamespaceAliasesOnly) == 0 && !originalBinder.InCrefButNotParameterOrReturnType)
             {
-                LookupMembersInInterfacesWithoutInheritance(current, type.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics), name, arity, basesBeingResolved, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics);
+                LookupMembersInInterfacesWithoutInheritance(current, GetBaseInterfaces(type, basesBeingResolved, ref useSiteDiagnostics),
+                    name, arity, basesBeingResolved, options, originalBinder, accessThroughType, diagnose, ref useSiteDiagnostics);
+            }
+
+        }
+
+        private static ImmutableArray<NamedTypeSymbol> GetBaseInterfaces(NamedTypeSymbol type, ConsList<TypeSymbol> basesBeingResolved, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            ImmutableArray<NamedTypeSymbol> baseInterfaces;
+            if (basesBeingResolved?.Any() != true)
+            {
+                baseInterfaces = type.AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics);
+            }
+            else
+            {
+                var interfacesLookedAt = new HashSet<NamedTypeSymbol>(TypeSymbol.EqualsConsiderEverything);
+                var baseInterfacesBuilder = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+
+                getBaseInterfaces(type, baseInterfacesBuilder, interfacesLookedAt, basesBeingResolved);
+
```

Comment:
I believe that `LookupMembersInInterfacesWithoutInheritance` depends for its correctness on its second parameter being topologically sorted. Please use Tarjan's algorithm to enumerate the base interfaces in topological order.